WordPress/gutenberg
# Blocks API: Add 'lock' attribute during block registration

## Description
Treat `lock` as a "build-in" attribute and include it during block registration. Removes filter previously used for adding this attribute.

Based on https://github.com/WordPress/gutenberg/pull/32457#discussion_r700917579.
Part of #29864.

## How has this been tested?
Check that unit tests are passing.

1. Use the snippet below to insert the locked block.
2. Block movers shouldn't be visible.
3. Remove block option isn't available.

## Example block
```html

<h2>Locked Heading</h2>
</div>
```

## Types of changes
Enhancement

## Checklist:
